libevmasm/AssemblyItem.h

```diff
@@ -196,22 +195,25 @@ class AssemblyItem
 	bool hasInstruction() const
 	{
 		return
-			m_type == Operation ||
-			m_type == EOFCreate ||
-			m_type == ReturnContract ||
-			m_type == RelativeJump ||
-			m_type == ConditionalRelativeJump ||
-			m_type == CallF ||
-			m_type == JumpF ||
-			m_type == RetF ||
-			m_type == SwapN ||
-			m_type == DupN;
+			m_instruction.has_value() &&
+			(
+				m_type == Operation ||
+				m_type == EOFCreate ||
+				m_type == ReturnContract ||
+				m_type == RelativeJump ||
+				m_type == ConditionalRelativeJump ||
+				m_type == CallF ||
+				m_type == JumpF ||
+				m_type == RetF ||
+				m_type == SwapN ||
+				m_type == DupN
+			);
```

Comment:
This actually changes behavior, because it lets you use some of these assembly items without an instruction. They are by design meant to always have one.

The added part should be an assert instead.

Reply:
Also for `verbatimBytecode`? The `m_instruction` used to be defaulted in that case (ie `STOP`), but I guess in reality it should signify some sort of null state, as in it's not used. I have now asserted that the `m_instruction` is set iff `hasInstruction` yields `true`. Also happy to change it to always assert `m_instruction.has_value()` in that function, although then the purpose of `hasInstruction` is a bit unclear to me, too :)

Reply:
> Also for `verbatimBytecode`?

Yes, it should not be used for verbatim. Verbatim carries a piece of bytecode in `data` instead.

> I have now asserted that the `m_instruction` is set iff `hasInstruction` yields `true`. 

Good. That's the intention. There are items that are associated with a single opcode and others that aren't. There's no in-between. We could clarify that in the docstring.

The instruction was originally used only with `Operation` but with EOF there are more items like this. Some of the existing ones could actually have an instruction too (e.g. the various push variants or `AuxDataLoadN`). We should refactor this at some point.